src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php

```diff
@@ -0,0 +1,195 @@
+            'Usage of `->will($this->returnValue(..))` statements must be replaced by it\'s shorter equivalent `->willReturn(...)`.',
+            [
+                new CodeSample('<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+    public function testSomeTest()
+    {
+        $someMock = $this->createMock(Some::class);
+        $someMock->method(\'some\')->will($this->returnSelf());
+        $someMock->method(\'some\')->will($this->returnValue(\'example\'));
+        $someMock->method(\'some\')->will($this->returnArgument(2));
+        $someMock->method(\'some\')->will($this->returnCallback(\'str_rot13\'));
+        $someMock->method(\'some\')->will($this->returnValueMap([\'a\',\'b\',\'c\']));
+    }
+}
+'),
+            ],
+            null,
+            'Risky when PHPUnit classes are overridden or not accessible, or when project has PHPUnit incompatibilities.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return !empty($tokens->findGivenKind(array_keys(static::RETURN_METHODS_MAP)));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder(static::METHOD_RETURN_VALUE, 'Fix `returnValue` occurrence'))
+                ->setAllowedTypes(['bool'])
+                ->setAllowedValues([false, true])
+                ->setDefault(true)
+                ->getOption(),
+            (new FixerOptionBuilder(static::METHOD_RETURN_SELF, 'Fix `returnSelf` occurrence'))
+                ->setAllowedTypes(['bool'])
+                ->setAllowedValues([false, true])
+                ->setDefault(true)
+                ->getOption(),
+            (new FixerOptionBuilder(static::METHOD_RETURN_ARGUMENT, 'Fix `returnArgument` occurrence'))
+                ->setAllowedTypes(['bool'])
+                ->setAllowedValues([false, true])
+                ->setDefault(true)
+                ->getOption(),
+            (new FixerOptionBuilder(static::METHOD_RETURN_CALLBACK, 'Fix `returnCallback` occurrence'))
+                ->setAllowedTypes(['bool'])
+                ->setAllowedValues([false, true])
+                ->setDefault(true)
+                ->getOption(),
+            (new FixerOptionBuilder(static::METHOD_RETURN_VALUE_MAP, 'Fix `returnValueMap` occurrence'))
+                ->setAllowedTypes(['bool'])
+                ->setAllowedValues([false, true])
+                ->setDefault(true)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($this->getSequences() as $method => $sequence) {
+            $this->fixSequence($tokens, $sequence, $method);
```

Comment:
basically, you iterate over whole collection 5 times. it;s slow.
iterate only once, by searching for sequence with `[T_STRING]` instead of `[T_STRING, $longerOccurrence]` and then see if given `T_STRING` matches method name you are interested in